cli/src/main/java/org/eclipse/hono/cli/client/ClientConfig.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli.client;
+
+import org.eclipse.hono.client.ApplicationClientFactory;
+import org.eclipse.hono.client.ConnectionLifecycle;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.client.device.amqp.AmqpAdapterClientFactory;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.springframework.util.StringUtils;
+
+import io.vertx.core.Vertx;
+
+/**
+ * Configuration class. Holds all the parameters to execute a command.
+ * <p>
+ * A default instance will be created at startup and it will be cloned and modified with custom
+ * parameter for the execution of a command.
+ */
+public class ClientConfig {
```

Comment:
there doesn't seem to be a real need for this class, is there? The default properties are passed in to the **Profile* classes via *ClientConfigProperties* and for the *connect** commands, you only need to create a copy of it and add the properties provided to the command as params (which you already do). From these updated *ClientConfigProperties* you can then create the client factory and the connection, right?

Reply:
Exactly, the commented description is an oversight.
I'm also thinking to refactor the name of the class. Since it holds only the profile and the factory for the active profile it's not a real `ClientConfig`... what do you say?

Reply:
My point is that I believe that we can safely remove this class as it seems to have no real purpose. In fact, FMPOV it makes things more complicated than they need to be ...

Reply:
Ok i see, since i do't have anymore a generalized connection method i don't need it. You are right, I have to pay more attention in the refactoring of the code.